File: backend/src/main/java/com/example/backend/domain/category/exception/CategoryException.java
package com.example.backend.domain.category.exception;

import org.springframework.http.HttpStatus;

public class CategoryException extends RuntimeException{

    private final CategoryErrorCode categoryErrorCode;

    public CategoryException(CategoryErrorCode categoryErrorCode) {
        super(categoryErrorCode.getMessage());
        this.categoryErrorCode = categoryErrorCode;
    }

    public HttpStatus getStatus() {
        return categoryErrorCode.getHttpStatus();
    }

    public String getCode() {
        return categoryErrorCode.getCode();
    }
}

File: backend/src/main/java/com/example/backend/domain/category/exception/CategoryException.kt
package com.example.backend.domain.category.exception

import org.springframework.http.HttpStatus

class CategoryException(val categoryErrorCode: CategoryErrorCode) : RuntimeException(categoryErrorCode.message) {

    override val message: String
        get() = categoryErrorCode.message

    fun getStatus(): HttpStatus = categoryErrorCode.httpStatus

    fun getCode(): String = categoryErrorCode.code
}
